src/V3Param.cpp

```diff
@@ -1003,6 +1003,8 @@ class ParamVisitor final : public VNVisitor {
     // Map from AstNodeModule to set of all AstNodeModules that instantiates it.
     std::unordered_map<AstNodeModule*, std::unordered_set<AstNodeModule*>> m_parentps;
 
+    bool m_inLparam = false;  // True if in localparam assignment
+
```

Comment:
I don't think this is the most efficient way to do this. I had v3Broken issues when doing cloneTree under something that wasn't a param assignment (t_interface_parameter_access.v) @RRozak @donlon  I saw you were editing v3param recently, do you guys want to take a look?

Reply:
Sure! So the cause of #3858 turn out to be `V3Const::constifyParamsEdit` in `ParamVisitor::visit(AstVar*)` corrupting the variables inside the module referenced, and it's before other parameterized instances (e.g. `SimpleIntf#(64) core_intf`) are cloned.

Cloning the variable expression does help but still has some limititions. I put a counterexample below.